### PR TITLE
{lang}[fosscuda/2019b] rstudio v1.2.5033, R v3.6.2

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

Alternative using Java 11: https://github.com/easybuilders/easybuild-easyconfigs/pull/9826

### CODE REVIEW

File: easybuild/easyconfigs/r/rstudio/rstudio-1.2.5033-fosscuda-2019b-Java-1.8.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+name = 'rstudio'
+version = '1.2.5033'
+versionsuffix = '-Java-%(javaver)s'
+
+homepage = 'https://www.rstudio.com/'
+description = """This RStudio Server version.
+RStudio is a set of integrated tools designed to help you be more productive with R.
+
+The server can be started with:  rserver --server-daemonize=0 --www-port 8787 --rsession-which-r=$(which R) --auth-none=1
```
Comment (terjekv) [inline easybuild/easyconfigs/r/rstudio/rstudio-1.2.5033-fosscuda-2019b-Java-1.8.eb]: Is there a specific reason why `--auth-none=1` is being suggested?

File: easybuild/easyconfigs/r/R/R-3.6.2-fosscuda-2019b-Java-1.8.eb [hunk 2]
```diff
@@ -0,0 +1,2356 @@
...
+        'checksums': ['466d428d25c066c9c96d892f24da930513d42b1bdf76d3b53628c3ba13c3e48a'],
+    }),
+    ('OceanView', '1.0.4', {
+        'checksums': ['e67f6f376737e1e5cf22fdfe2769a8f674e90c886b0e43942e97d9a3e6924f1c'],
+    }),
+    ('ks', '1.11.6', {
+        'checksums': ['d8f1ccb99281eed092b8f950cfbf422330e88f2a0ce1ef6c9dbd7d8b9d648c41'],
+    }),
+    ('logcondens', '2.1.5', {
+
```
Comment (terjekv) [inline easybuild/easyconfigs/r/R/R-3.6.2-fosscuda-2019b-Java-1.8.eb]: Could we also get RStan in here? As per https://github.com/easybuilders/easybuild-easyconfigs/pull/9758 ? 

`('loo', '2.2.0', {
        'checksums': ['466df60953a89fcb135b32909197c3ff26ecea719c191667faa5747324fb01c3'],
    }),
    ('StanHeaders', '2.21.0-1', {
        'checksums': ['4e94148af2960f203b208c2b725d8be628ca282c7a1e967a7e6e78aa5eb90d3f'],
    }),
    ('rstan', '2.19.2', {
        'checksums': ['31e4ceb9c327cd62873225097ffa538c2ac4cb0547c52271e52e4c7652d508da'],
    }),`

File: easybuild/easyconfigs/r/rstudio/rstudio-1.2.5033-fosscuda-2019b-Java-1.8.eb (hunk 1 above)
Comment (Flamefire) [reply easybuild/easyconfigs/r/rstudio/rstudio-1.2.5033-fosscuda-2019b-Java-1.8.eb]: This was suggested by a user. I think I'll remove it for the official EC

File: easybuild/easyconfigs/r/R/R-3.6.2-fosscuda-2019b-Java-1.8.eb (hunk 2 above)
Comment (Flamefire) [reply easybuild/easyconfigs/r/R/R-3.6.2-fosscuda-2019b-Java-1.8.eb]: Sure

Comment (terjekv) [reply easybuild/easyconfigs/r/R/R-3.6.2-fosscuda-2019b-Java-1.8.eb]: Thanks!

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 5 out of 7 (2 easyconfigs in this PR)
ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/55f47b39d14d6d3ded0e9a6261580f04 for a full test report.

**Comment by terjekv** [comment]
Test report by @terjekv
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/cca5c8eed4b6a2a29273f07528882b21 for a full test report.

**Comment by terjekv** [comment]
> Test report by @terjekv
> **FAILED**
> Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
> ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
> See https://gist.github.com/cca5c8eed4b6a2a29273f07528882b21 for a full test report.

It's getting to the point where testing builds with R is doomed to fail...

**Comment by terjekv** [comment]
Test report by @terjekv
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
ninhursaga.uio.no - Linux red hat enterprise linux 8.1, Intel(R) Xeon(R) Silver 4216 CPU @ 2.10GHz, Python 3.6.8
See https://gist.github.com/037ec5c2251e7d073275a029f4bc04d1 for a full test report.

**Comment by Flamefire** [comment]
Yeah. I Updated the checksums now with force-download option. But I guess we should rather disable checksums for CRAN until they fix their stuff

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/ff2f0e8d5113a87f23a29353a11ab73b for a full test report.

**Comment by Flamefire** [comment]
@verdurin Please rerun with disabled checksum verification... CRAN again -.-

**Comment by verdurin** [comment]
@Flamefire yes, I saw. Could you at least add Kenneth's latest updates from https://github.com/easybuilders/easybuild-easyconfigs/commit/23a63f9f4a5a2a00070260eb74289bfac093e2c4 ?

**Comment by Flamefire** [comment]
Done by Copy-Pasting the stuff from fosscuda2019b (made by @boegel )

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/7becd4526061a88f5b4e956006fa5eb1 for a full test report.

**Comment by Flamefire** [comment]
@verdurin 
>   libcuda.so.1: cannot open shared object file: No such file or directory

Is your CUDA module broken or does not set LD_LIBRARY_PATH? Strange that it compiled but does not run.

**Comment by edmondac** [comment]
I believe this patch makes it work with Java 11

**rstudio-1.2.5033_update_gwt.patch**
```
--- a/dependencies/common/install-gwt   2020-02-04 11:16:49.907923640 +0000
+++ b/dependencies/common/install-gwt   2020-02-04 11:16:38.932802049 +0000
@@ -61,7 +61,7 @@
 fi
 
 # gwt sdk
-GWT_SDK_VER=2.8.1
+GWT_SDK_VER=2.8.2
 GWT_SDK=gwt-$GWT_SDK_VER
 GWT_SDK_ZIP=$GWT_SDK.zip
 if [ -d "$LIB_DIR/gwt/$GWT_SDK_VER" ]
--- a/src/gwt/build.xml 2020-02-04 11:24:38.224107923 +0000
+++ b/src/gwt/build.xml 2020-02-04 11:24:57.542321679 +0000
@@ -29,7 +29,7 @@
             classpath="${tools.dir}/compiler/compiler.jar"/>
 
    <!-- Configure path to GWT SDK -->
-   <property name="gwt.sdk" value="${lib.dir}/gwt/2.8.1"/>
+   <property name="gwt.sdk" value="${lib.dir}/gwt/2.8.2"/>
 
    <property name="gwt.extra.args" value=""/>
    <property name="gwt.main.module" value="org.rstudio.studio.RStudio"/>
```

Still testing it, but it certainly builds ok...

**Comment by edmondac** [comment]
That PR is an internal one for our team, but it's got the Java 11 version of this. Still testing it...

**Comment by Flamefire** [comment]
Summarizing conversation with @edmondac 

- Current RStudio uses GWT 2.8.1 which it downloads during compilation
- GWT 2.8.**2** is (somewhat?) compatible with Java 9: "Supports running in Java 9. Note that this does not yet mean that GWT can compile Java 9 sources, or support the Java 9 JRE changes, but that a Java 9 JRE can be used to compile a GWT project."
- GWT 2.9 may support Java 11: https://github.com/gwtproject/gwt/issues/9626
- RStudio does not support Java > 8: https://github.com/rstudio/rstudio/wiki/Installing-RStudio-Dependencies "You can install Java 8 from the following URL (note that Java 9 and 10 are currently unsupported):"
- @edmondac reports that using GWT 2.8.2 RStudio can be compiled with Java 11 and starts without issues. 
  - "gwt is building with source code version 1.8, so it's not a matter of supporting java 11 code " but just fixing a bug which makes it incompatible with Java 9+

I'm very reluctant in adding an "official" EC for an officially unsupported configuration but IMO not having another flavor of R would make it worth trying. I'll give it a shot on our x86 and Power9 nodes and see what happens

File: easybuild/easyconfigs/r/rstudio/rstudio-1.2.5033-fosscuda-2019b-Java-1.8.eb (hunk 1 above)
Comment (lazariv) [reply easybuild/easyconfigs/r/rstudio/rstudio-1.2.5033-fosscuda-2019b-Java-1.8.eb]: > Is there a specific reason why `--auth-none=1` is being suggested?

The reason is that the open-source version of RStudio Server does not support any other authentification method (such as e.g. LDAP, which is currently used on our system) except the system users, created with `useradd` .

Comment (terjekv) [reply easybuild/easyconfigs/r/rstudio/rstudio-1.2.5033-fosscuda-2019b-Java-1.8.eb]: @lazariv Ah, that makes sense, but how does RStudio auth actually resolve system users? Do they need to be in /etc/passwd, or does it work with sss_cache and friends that then get their data from LDAP? Those users are able to log into the machine.

Comment (lazariv) [reply easybuild/easyconfigs/r/rstudio/rstudio-1.2.5033-fosscuda-2019b-Java-1.8.eb]: @terjekv As far as I know, it uses PAM for authentification.  Some more information is available here [https://jstaf.github.io/2018/06/20/rstudio-server-semi-pro.html](url)

Comment (terjekv) [reply easybuild/easyconfigs/r/rstudio/rstudio-1.2.5033-fosscuda-2019b-Java-1.8.eb]: https://jstaf.github.io/2018/06/20/rstudio-server-semi-pro.html was quite helpful, thanks! Although I'm not sure I want to work on PAM with something coming from modules. :-)

**Comment by Flamefire** [comment]
Closing in favor of #9826

**Pull request closed without merging**